GraphConsoleAppV3/GraphRequests.cs

```diff
@@ -0,0 +1,1534 @@
+            Console.WriteLine("\nGetting Devices");
+            IPagedCollection<IDevice> devices = null;
+            try
+            {
+                devices = await client.Devices.ExecuteAsync();
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine("/nError getting devices {0} {1}", e.Message,
+                    e.InnerException != null ? e.InnerException.Message : "");
+            }
+
+            if (devices != null)
+            {
+                do
+                {
+                    List<IDevice> devicesList = devices.CurrentPage.ToList();
+                    foreach (IDevice device in devicesList)
+                    {
+                        if (device.ObjectId != null)
+                        {
+                            Console.WriteLine("Device ID: {0}, Type: {1}", device.DeviceId, device.DeviceOSType);
+                            IPagedCollection<IDirectoryObject> registeredOwners = device.RegisteredOwners;
+                            if (registeredOwners != null)
+                            {
+                                do
+                                {
+                                    List<IDirectoryObject> registeredOwnersList = registeredOwners.CurrentPage.ToList();
+                                    foreach (IDirectoryObject owner in registeredOwnersList)
+                                    {
+                                        Console.WriteLine("Device Owner ID: " + owner.ObjectId);
+                                    }
+                                    registeredOwners = await registeredOwners.GetNextPageAsync();
+                                } while (registeredOwners != null);
+                            }
+                        }
+                    }
+                    devices = await devices.GetNextPageAsync();
+                } while (devices != null);
+            }
+
+            #endregion
+        }
+
+        private static async Task CreateOAuth2Permission(IActiveDirectoryClient client, ServicePrincipal newServicePrincpal)
+        {
+            #region Create a new consentable OAuth2 permission
+
+            //*********************************************************************************************
+            // Create new  oauth2 permission object
+            //*********************************************************************************************
+            if (newServicePrincpal.ObjectId != null)
+            {
+                OAuth2PermissionGrant permissionObject = new OAuth2PermissionGrant();
+                permissionObject.ConsentType = "AllPrincipals";
+                permissionObject.Scope = "user_impersonation";
+                permissionObject.StartTime = DateTime.Now;
+                permissionObject.ExpiryTime = (DateTime.Now).AddMonths(12);
+
+                // resourceId is objectId of the resource, in this case objectId of AzureAd (Graph API)
```

Comment:
>objectId of AzureAd [](start = 72, length = 19)

You might want to add clearer comment since it didnt make sense to you when you were trying to make it work.